components/plurals/src/rules/runtime/ast.rs

```diff
@@ -14,17 +14,20 @@ use zerovec::{
 };
 
 #[derive(yoke::Yokeable, zerofrom::ZeroFrom, Clone, PartialEq, Debug)]
+#[allow(clippy::exhaustive_structs)] // this type is stable
 pub struct Rule<'data>(pub VarZeroVec<'data, RelationULE>);
```

Comment:
Suggestion: The VarZeroVec should be a private field, and all the other types in this module should be pub(crate).